### PR TITLE
syncer: add real binlog pos to log

### LINKED ISSUE #24: write mysql binlog next position information in log.
Now we only write save position, which is not the same as the master mysql position from `show master status`, we should add a new trace variable to log this.

### PR DESCRIPTION
**Pull request by andelf**
Fix #24 .

- add new job type ``xid``
- add a ``realPos`` field to ``Syncer``
- add an extra filed to status log

### CODE REVIEW

**Comment by qiuyesuifeng** [comment]
LGTM

File: syncer/syncer.go [hunk 1]
```diff
@@ -364,16 +364,21 @@ func (s *Syncer) checkWait(job *job) bool {
 }
 
 func (s *Syncer) addJob(job *job) error {
-	s.jobWg.Add(1)
+	if job.tp == xid {
+		s.meta.Save(job.pos, false)
```
Comment (nolouch) [inline syncer/syncer.go]: why need force set false? seems if restart it will repeat some event from last position

Comment (andelf) [inline syncer/syncer.go]: Saving meta data to disk for every XID event leads to bad performance.
Repeating some events is the expected behaviour of syncer.
refer: [sql insert is translated to replace info](https://github.com/pingcap/tidb-tools/blob/master/syncer/db.go#L196)

**Comment by qiuyesuifeng** [comment]
@siddontang @nolouch  PTAL

**Comment by nolouch** [comment]
LGTM

File: syncer/syncer.go (hunk 1 above)
Comment (nolouch) [inline syncer/syncer.go]: maybe we can save position again when quit

**Comment by siddontang** [comment]
> maybe we can save position again when quit

Do we save when quit now?

**Comment by qiuyesuifeng** [comment]
We can save before quit, but it is not necessary i think.

**Pull request merged**